docs/public/_redirects

```diff
@@ -20,10 +20,9 @@
 /r/issue-template-latest https://codesandbox.io/s/material-ui-issue-latest-s2dsx
 /r/ts-issue-template https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQokscA3nXHAPSdwwAWWOLhKQAdllEx0ATwgBXOHNRYAJnQC+cIiXIABEHOCcQyGFijBkAGzJ06BOaPzAIouABEsICAAoAlKzsXDwAmvJQQhAqWBpAA 302
 /r/custom-component-variants /customization/components/#adding-new-component-variants
-/r/x-license https://mui.com/store/items/material-ui-pro/
```

Comment:
We are using the /r/ folder anytime we need to link a URL in the code. This is because we can't update the code that is shipped to the users. A systematic redirection brings two values:

1. We can link anchors
2. We are less likely to forget updating a 301 hence, having a 404 from the content linked in the code.

You can see this used in https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react-reconciler/src/ReactStrictModeWarnings.new.js#L196.


This is the case for us in https://github.com/mui/mui-x/blob/ebb0e3b1a4d940b02ca20abd61ade86fce9b6e94/packages/x-license-pro/LICENSE#L6. So I think that we should keep this line and replace the link used in the license files.

Reply:
> Should we then update /r/x-license to https://mui.com/store/legal/mui-x-eula/ ?

Yes, I think that it would be perfect